mutation-testing/output-pr-mutants/action.yml

```diff
@@ -0,0 +1,166 @@
+              echo "</details>" >> "$GITHUB_STEP_SUMMARY"
+              echo "" >> "$GITHUB_STEP_SUMMARY"
+            fi
+
+            if [[ "$section_title" != "" ]]; then
+              awk -F':' '{printf "- [ ] " "[" $0 "]"; file_path=$1; line=$2; $1=""; $2=""; printf "(" "'"$server_url"'/'"$organisation"'/'"$repository"'/blob/'"$commit"'/" file_path "#L" line-1 ")\n\n"}' "$file_name" >> "$GITHUB_STEP_SUMMARY"
+            else
+              awk -F':' '{printf "- [x] " "[" $0 "]"; file_path=$1; line=$2; $1=""; $2=""; printf "(" "'"$server_url"'/'"$organisation"'/'"$repository"'/blob/'"$commit"'/" file_path "#L" line-1 ")\n\n"}' "$file_name" >> "$GITHUB_STEP_SUMMARY"
+            fi
+
+            if [[ "$section_title" == "Missed:" ]]; then
+              echo "### To resolve this issue, consider one of the following options:" >> "$GITHUB_STEP_SUMMARY"
+              echo "- Modify or add tests including this function." >> "$GITHUB_STEP_SUMMARY"
+              echo "- If you are absolutely certain that this function should not undergo mutation testing, add '#[mutants::skip]' or '#[cfg_attr(test, mutants::skip)]' function header to skip it." >> "$GITHUB_STEP_SUMMARY"
+            elif [[ "$section_title" == "Timeout:" ]]; then
+              echo "### To resolve this issue, consider one of the following options:" >> "$GITHUB_STEP_SUMMARY"
+              echo "- Modify the tests that include this funcion." >> "$GITHUB_STEP_SUMMARY"
+              echo "- Add '#[mutants::skip]' or '#[cfg_attr(test, mutants::skip)]' function header to skip it." >> "$GITHUB_STEP_SUMMARY"
+            elif [[ "$section_title" == "Unviable:" ]]; then
+              echo "### To resolve this issue, consider one of the following options:" >> "$GITHUB_STEP_SUMMARY"
+              echo "- Create 'Default::default()' implementation for the specific structure." >> "$GITHUB_STEP_SUMMARY"
+              echo "- Add '#[mutants::skip]' or '#[cfg_attr(test, mutants::skip)]' function header to skip it." >> "$GITHUB_STEP_SUMMARY"
+            fi
+
+            echo >> "$GITHUB_STEP_SUMMARY"
+          fi
+        }
+
+        # Print uncaught (missed/timeout/unviable) mutants to summary
+        echo "# Uncaught Mutants" >> "$GITHUB_STEP_SUMMARY"
+        write_section "Missed:" "./mutants-shards/missed.txt"
+        write_section "Timeout:" "./mutants-shards/timeout.txt"
+        write_section "Unviable:" "./mutants-shards/unviable.txt"
+
+        # Print caught mutants to summary
+        echo "# Caught Mutants" >> "$GITHUB_STEP_SUMMARY"
+        write_section "" "./mutants-shards/caught.txt"
+
+        # Get most relevant exit code from the file and match it
+        exit_code=$(<"mutants-shards/exit_code.txt")
+
+        case $exit_code in
+          0)
+              if [[ -f ./mutants-shards/unviable.txt ]]; then
+                echo "Found unviable mutants!"
+                exit 1
```

Comment:
```suggestion
                exit $exit_code
```

Reply:
I will have to hardcode an exit code 5 here. In the case there are only unviable mutants, the `cargo mutants` command exits with code 0, which is the same as the case where all mutants are caught. That is why i am testing for unviable before proceeding to exit successfully.

Reply:
strange, but understood - i've seen similar behaviour where errors/success both exit with 0 and it's annoying